tests/rptest/tests/internal_topic_protection_test.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import subprocess
+import time
+
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.rpk import RpkTool, RpkException
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.kafka_cat import KafkaCat
+from ducktape.utils.util import wait_until
+
+
+class InternalTopicProtectionTest(RedpandaTest):
+    """
+    Verify that the `kafka_nodelete_topics` and `kafka_noproduce_topics`
+    configuration properties function as intended.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, extra_rp_conf={}, **kwargs)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.kafka_cat = KafkaCat(self.redpanda)
+        self.rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=3)
+    def kafka_protections_disable_config_test(self):
+        def try_config_change(
+                test_topic,
+                protect_config,
+                config="retention.ms",
+                original_val=str(4 * 60 * 60 * 60),  # 4 hrs
+                new_val=str(5 * 60 * 60 * 60)  # 5 hrs
+        ):
+
+            self.rpk.create_topic(test_topic, 3, config={config: original_val})
+
+            # Protect topic
+            self.rpk.cluster_config_set(protect_config, str([test_topic]))
+
+            # Ensure config of protected topic can't be changed.
+            self.rpk.alter_topic_config(test_topic, config, new_val)
+
+            # Allow time for a potential change to be propagated.
+            time.sleep(10)
+            config_val, _ = self.rpk.describe_topic_configs(test_topic)[config]
+
+            assert config_val == original_val, "Topic config was changed even with protection"
+
+            # Ensure config of protected topic can't be deleted.
+            self.rpk.delete_topic_config(test_topic, config)
+
+            # Allow time for a potential change to be propagated.
+            time.sleep(10)
+            config_val, _ = self.rpk.describe_topic_configs(test_topic)[config]
+
+            assert config_val == original_val, "Topic config was deleted even with protection"
+
+        try_config_change("no_delete_topic", "kafka_nodelete_topics")
+        try_config_change("no_produce_topic", "kafka_noproduce_topics")
```

Comment:
Let's add a couple lines to each of these tests that removes the topic from the nodelete/noproduce list with RedpandaService.set_cluster_conf, and then confirms that the action starts working.

That way we're validating that our escape route works in case user wants to force it, and also that our delete/produce functions were really failing because of our nodelete/noproducer settings, and not some other reason.